Unable to compile in PlatformIO for Arduino/Teensy 4.x

### LVGL version

v9.0.0

### What happened?

I am trying to build v9.0.0 with the music demo, and I have added "Arudino.h" / "avr/pgmspace.h" to use my PROGMEM and EXTMEM marcos to put big constant arrays into program memory and big arrays into SDRAM - I do this with v8.3.x and it works fine, but on v9.0.0 I get the following compile errors:

``` /Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h: Assembler messages:
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:41: Error: bad instruction 'typedef signed char __int8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:43: Error: bad instruction 'typedef unsigned char __uint8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:55: Error: bad instruction 'typedef short int __int16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:57: Error: bad instruction 'typedef short unsigned int __uint16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:77: Error: bad instruction 'typedef long int __int32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:79: Error: bad instruction 'typedef long unsigned int __uint32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:103: Error: bad instruction 'typedef long long int __int64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:105: Error: bad instruction 'typedef long long unsigned int __uint64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:134: Error: bad instruction 'typedef signed char __int_least8_t'
Compiling .pio/build/teensymm/src/lvgl/src/draw/sw/lv_draw_sw_fill.c.o
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:136: Error: bad instruction 'typedef unsigned char __uint_least8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:160: Error: bad instruction 'typedef short int __int_least16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:162: Error: bad instruction 'typedef short unsigned int __uint_least16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:182: Error: bad instruction 'typedef long int __int_least32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:184: Error: bad instruction 'typedef long unsigned int __uint_least32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:200: Error: bad instruction 'typedef long long int __int_least64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:202: Error: bad instruction 'typedef long long unsigned int __uint_least64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:214: Error: bad instruction 'typedef long long int __intmax_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:222: Error: bad instruction 'typedef long long unsigned int __uintmax_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:230: Error: bad instruction 'typedef int __intptr_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/machine/_default_types.h:232: Error: bad instruction 'typedef unsigned int __uintptr_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:20: Error: bad instruction 'typedef __int8_t int8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:24: Error: bad instruction 'typedef __uint8_t uint8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:32: Error: bad instruction 'typedef __int16_t int16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:36: Error: bad instruction 'typedef __uint16_t uint16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:44: Error: bad instruction 'typedef __int32_t int32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:48: Error: bad instruction 'typedef __uint32_t uint32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:56: Error: bad instruction 'typedef __int64_t int64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:60: Error: bad instruction 'typedef __uint64_t uint64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:67: Error: bad instruction 'typedef __intmax_t intmax_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:72: Error: bad instruction 'typedef __uintmax_t uintmax_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:77: Error: bad instruction 'typedef __intptr_t intptr_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_stdint.h:82: Error: bad instruction 'typedef __uintptr_t uintptr_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:21: Error: bad instruction 'typedef __int_least8_t int_least8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:22: Error: bad instruction 'typedef __uint_least8_t uint_least8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:27: Error: bad instruction 'typedef __int_least16_t int_least16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:28: Error: bad instruction 'typedef __uint_least16_t uint_least16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:33: Error: bad instruction 'typedef __int_least32_t int_least32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:34: Error: bad instruction 'typedef __uint_least32_t uint_least32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:39: Error: bad instruction 'typedef __int_least64_t int_least64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:40: Error: bad instruction 'typedef __uint_least64_t uint_least64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:51: Error: bad instruction 'typedef int int_fast8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:52: Error: bad instruction 'typedef unsigned int uint_fast8_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:61: Error: bad instruction 'typedef int int_fast16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:62: Error: bad instruction 'typedef unsigned int uint_fast16_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:71: Error: bad instruction 'typedef int int_fast32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:72: Error: bad instruction 'typedef unsigned int uint_fast32_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:81: Error: bad instruction 'typedef long long int int_fast64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/stdint.h:82: Error: bad instruction 'typedef long long unsigned int uint_fast64_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/lib/gcc/arm-none-eabi/11.3.1/include/stddef.h:321: Error: bad instruction 'typedef unsigned int wchar_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_locale.h:9: Error: bad instruction 'struct __locale_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/sys/_locale.h:10: Error: bad instruction 'typedef struct __locale_t*locale_t'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:312: Error: bad instruction 'typedef struct {'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:313: Error: bad instruction 'intmax_t quot'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:314: Error: bad instruction 'intmax_t rem'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:315: Error: junk at end of line, first unrecognized character is '}'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:317: Error: bad instruction 'struct _reent'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:323: Error: bad instruction 'extern intmax_t imaxabs(intmax_t)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:324: Error: bad instruction 'extern imaxdiv_t imaxdiv(intmax_t __numer,intmax_t __denomer)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:325: Error: bad instruction 'extern intmax_t strtoimax(const char*__restrict,char**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:326: Error: bad instruction 'extern intmax_t _strtoimax_r(struct _reent*,const char*__restrict,char**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:327: Error: bad instruction 'extern uintmax_t strtoumax(const char*__restrict,char**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:328: Error: bad instruction 'extern uintmax_t _strtoumax_r(struct _reent*,const char*__restrict,char**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:329: Error: bad instruction 'extern intmax_t wcstoimax(const wchar_t*__restrict,wchar_t**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:330: Error: bad instruction 'extern intmax_t _wcstoimax_r(struct _reent*,const wchar_t*__restrict,wchar_t**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:331: Error: bad instruction 'extern uintmax_t wcstoumax(const wchar_t*__restrict,wchar_t**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:332: Error: bad instruction 'extern uintmax_t _wcstoumax_r(struct _reent*,const wchar_t*__restrict,wchar_t**__restrict,int)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:335: Error: bad instruction 'extern intmax_t strtoimax_l(const char*__restrict,char**_restrict,int,locale_t)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:336: Error: bad instruction 'extern uintmax_t strtoumax_l(const char*__restrict,char**_restrict,int,locale_t)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:337: Error: bad instruction 'extern intmax_t wcstoimax_l(const wchar_t*__restrict,wchar_t**_restrict,int,locale_t)'
/Users/user-xyz/.platformio/packages/toolchain-gccarmnoneeabi-teensy/arm-none-eabi/include/inttypes.h:338: Error: bad instruction 'extern uintmax_t wcstoumax_l(const wchar_t*__restrict,wchar_t**_restrict,int,locale_t)'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:39: Error: bad instruction 'typedef void prog_void'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:40: Error: bad instruction 'typedef char prog_char'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:41: Error: bad instruction 'typedef unsigned char prog_uchar'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:42: Error: bad instruction 'typedef int8_t prog_int8_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:43: Error: bad instruction 'typedef uint8_t prog_uint8_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:44: Error: bad instruction 'typedef int16_t prog_int16_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:45: Error: bad instruction 'typedef uint16_t prog_uint16_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:46: Error: bad instruction 'typedef int32_t prog_int32_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:47: Error: bad instruction 'typedef uint32_t prog_uint32_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:48: Error: bad instruction 'typedef int64_t prog_int64_t'
/Users/user-xyz/.platformio/packages/framework-arduinoteensy/cores/teensy4/avr/pgmspace.h:49: Error: bad instruction 'typedef uint64_t prog_uint64_t'
*** [.pio/build/teensymm/src/lvgl/src/draw/sw/blend/neon/lv_blend_neon.S.o] Error 1
```

I need these macros otherwise I get ram overflow errors

### How to reproduce?

``` /**
 * @file lv_conf.h
 * Configuration file for v9.0.0
 */

/*
 * Copy this file as `lv_conf.h`
 * 1. simply next to the `lvgl` folder
 * 2. or any other places and
 *    - define `LV_CONF_INCLUDE_SIMPLE`
 *    - add the path as include path
 */

/* clang-format off */
#if 1 /*Set it to "1" to enable content*/

#ifndef LV_CONF_H
#define LV_CONF_H

#include <avr/pgmspace.h>
//#include "Arduino.h"

/*====================
   COLOR SETTINGS
 *====================*/

/*Color depth: 8 (A8), 16 (RGB565), 24 (RGB888), 32 (XRGB8888)*/
#define LV_COLOR_DEPTH 24

/*=========================
   STDLIB WRAPPER SETTINGS
 *=========================*/

/* Possible values
 * - LV_STDLIB_BUILTIN:     LVGL's built in implementation
 * - LV_STDLIB_CLIB:        Standard C functions, like malloc, strlen, etc
 * - LV_STDLIB_MICROPYTHON: MicroPython implementation
 * - LV_STDLIB_RTTHREAD:    RT-Thread implementation
 * - LV_STDLIB_CUSTOM:      Implement the functions externally
 */
#define LV_USE_STDLIB_MALLOC    LV_STDLIB_BUILTIN
#define LV_USE_STDLIB_STRING    LV_STDLIB_BUILTIN
#define LV_USE_STDLIB_SPRINTF   LV_STDLIB_BUILTIN


#if LV_USE_STDLIB_MALLOC == LV_STDLIB_BUILTIN
    /*Size of the memory available for `lv_malloc()` in bytes (>= 2kB)*/
    #define LV_MEM_SIZE (128 * 1024U)          /*[bytes]*/

    /*Size of the memory expand for `lv_malloc()` in bytes*/
    #define LV_MEM_POOL_EXPAND_SIZE 0

    /*Set an address for the memory pool instead of allocating it as a normal array. Can be in external SRAM too.*/
    #define LV_MEM_ADR 0     /*0: unused*/
    /*Instead of an address give a memory allocator that will be called to get a memory pool for LVGL. E.g. my_malloc*/
    #if LV_MEM_ADR == 0
        #undef LV_MEM_POOL_INCLUDE
        #undef LV_MEM_POOL_ALLOC
    #endif
#endif  /*LV_USE_MALLOC == LV_STDLIB_BUILTIN*/

/*====================
   HAL SETTINGS
 *====================*/

/*Default display refresh, input device read and animation step period.*/
#define LV_DEF_REFR_PERIOD  17      /*[ms]*/

/*Default Dot Per Inch. Used to initialize default sizes such as widgets sized, style paddings.
 *(Not so important, you can adjust it to modify default sizes and spaces)*/
#define LV_DPI_DEF 130     /*[px/inch]*/

/*=================
 * OPERATING SYSTEM
 *=================*/
/*Select an operating system to use. Possible options:
 * - LV_OS_NONE
 * - LV_OS_PTHREAD
 * - LV_OS_FREERTOS
 * - LV_OS_CMSIS_RTOS2
 * - LV_OS_RTTHREAD
 * - LV_OS_WINDOWS
 * - LV_OS_CUSTOM */
#define LV_USE_OS   LV_OS_NONE

#if LV_USE_OS == LV_OS_CUSTOM
    #define LV_OS_CUSTOM_INCLUDE <stdint.h>
#endif

/*========================
 * RENDERING CONFIGURATION
 *========================*/

/*Align the stride of all layers and images to this bytes*/
#define LV_DRAW_BUF_STRIDE_ALIGN                1

/*Align the start address of draw_buf addresses to this bytes*/
#define LV_DRAW_BUF_ALIGN                       4

#define LV_USE_DRAW_SW 1
#if LV_USE_DRAW_SW == 1
    /* Set the number of draw unit.
     * > 1 requires an operating system enabled in `LV_USE_OS`
     * > 1 means multiply threads will render the screen in parallel */
    #define LV_DRAW_SW_DRAW_UNIT_CNT    1

    /* Use Arm-2D to accelerate the sw render */
    #define LV_USE_DRAW_ARM2D_SYNC      0

    /* If a widget has `style_opa < 255` (not `bg_opa`, `text_opa` etc) or not NORMAL blend mode
     * it is buffered into a "simple" layer before rendering. The widget can be buffered in smaller chunks.
     * "Transformed layers" (if `transform_angle/zoom` are set) use larger buffers
     * and can't be drawn in chunks. */

    /*The target buffer size for simple layer chunks.*/
    #define LV_DRAW_SW_LAYER_SIMPLE_BUF_SIZE    (24 * 1024)   /*[bytes]*/

    /* 0: use a simple renderer capable of drawing only simple rectangles with gradient, images, texts, and straight lines only
     * 1: use a complex renderer capable of drawing rounded corners, shadow, skew lines, and arcs too */
    #define LV_DRAW_SW_COMPLEX          1

    #if LV_DRAW_SW_COMPLEX == 1
        /*Allow buffering some shadow calculation.
        *LV_DRAW_SW_SHADOW_CACHE_SIZE is the max. shadow size to buffer, where shadow size is `shadow_width + radius`
        *Caching has LV_DRAW_SW_SHADOW_CACHE_SIZE^2 RAM cost*/
        #define LV_DRAW_SW_SHADOW_CACHE_SIZE 0

        /* Set number of maximally cached circle data.
        * The circumference of 1/4 circle are saved for anti-aliasing
        * radius * 4 bytes are used per circle (the most often used radiuses are saved)
        * 0: to disable caching */
        #define LV_DRAW_SW_CIRCLE_CACHE_SIZE 4
    #endif

    #define  LV_USE_DRAW_SW_ASM     LV_DRAW_SW_ASM_NONE

    #if LV_USE_DRAW_SW_ASM == LV_DRAW_SW_ASM_CUSTOM
        #define  LV_DRAW_SW_ASM_CUSTOM_INCLUDE ""
    #endif
#endif

/* Use NXP's VG-Lite GPU on iMX RTxxx platforms. */
#define LV_USE_DRAW_VGLITE 0

#if LV_USE_DRAW_VGLITE
    /* Enable blit quality degradation workaround recommended for screen's dimension > 352 pixels. */
    #define LV_USE_VGLITE_BLIT_SPLIT 0

    #if LV_USE_OS
        /* Enable VGLite draw async. Queue multiple tasks and flash them once to the GPU. */
        #define LV_USE_VGLITE_DRAW_ASYNC 1
    #endif

    /* Enable VGLite asserts. */
    #define LV_USE_VGLITE_ASSERT 0
#endif

/* Use NXP's PXP on iMX RTxxx platforms. */
#define LV_USE_DRAW_PXP 0

#if LV_USE_DRAW_PXP
    /* Enable PXP asserts. */
    #define LV_USE_PXP_ASSERT 0
#endif

/* Use Renesas Dave2D on RA  platforms. */
#define LV_USE_DRAW_DAVE2D 0

/* Draw using cached SDL textures*/
#define LV_USE_DRAW_SDL 0

/* Use VG-Lite GPU. */
#define LV_USE_DRAW_VG_LITE 0

#if LV_USE_DRAW_VG_LITE
/* Enable VG-Lite custom external 'gpu_init()' function */
#define LV_VG_LITE_USE_GPU_INIT 0

/* Enable VG-Lite assert. */
#define LV_VG_LITE_USE_ASSERT 0

#endif

/*=======================
 * FEATURE CONFIGURATION
 *=======================*/

/*-------------
 * Logging
 *-----------*/

/*Enable the log module*/
#define LV_USE_LOG 0
#if LV_USE_LOG

    /*How important log should be added:
    *LV_LOG_LEVEL_TRACE       A lot of logs to give detailed information
    *LV_LOG_LEVEL_INFO        Log important events
    *LV_LOG_LEVEL_WARN        Log if something unwanted happened but didn't cause a problem
    *LV_LOG_LEVEL_ERROR       Only critical issue, when the system may fail
    *LV_LOG_LEVEL_USER        Only logs added by the user
    *LV_LOG_LEVEL_NONE        Do not log anything*/
    #define LV_LOG_LEVEL LV_LOG_LEVEL_WARN

    /*1: Print the log with 'printf';
    *0: User need to register a callback with `lv_log_register_print_cb()`*/
    #define LV_LOG_PRINTF 0

    /*1: Enable print timestamp;
     *0: Disable print timestamp*/
    #define LV_LOG_USE_TIMESTAMP 1

    /*1: Print file and line number of the log;
     *0: Do not print file and line number of the log*/
    #define LV_LOG_USE_FILE_LINE 1

    /*Enable/disable LV_LOG_TRACE in modules that produces a huge number of logs*/
    #define LV_LOG_TRACE_MEM        1
    #define LV_LOG_TRACE_TIMER      1
    #define LV_LOG_TRACE_INDEV      1
    #define LV_LOG_TRACE_DISP_REFR  1
    #define LV_LOG_TRACE_EVENT      1
    #define LV_LOG_TRACE_OBJ_CREATE 1
    #define LV_LOG_TRACE_LAYOUT     1
    #define LV_LOG_TRACE_ANIM       1
    #define LV_LOG_TRACE_CACHE      1

#endif  /*LV_USE_LOG*/

/*-------------
 * Asserts
 *-----------*/

/*Enable asserts if an operation is failed or an invalid data is found.
 *If LV_USE_LOG is enabled an error message will be printed on failure*/
#define LV_USE_ASSERT_NULL          1   /*Check if the parameter is NULL. (Very fast, recommended)*/
#define LV_USE_ASSERT_MALLOC        1   /*Checks is the memory is successfully allocated or no. (Very fast, recommended)*/
#define LV_USE_ASSERT_STYLE         0   /*Check if the styles are properly initialized. (Very fast, recommended)*/
#define LV_USE_ASSERT_MEM_INTEGRITY 0   /*Check the integrity of `lv_mem` after critical operations. (Slow)*/
#define LV_USE_ASSERT_OBJ           0   /*Check the object's type and existence (e.g. not deleted). (Slow)*/

/*Add a custom handler when assert happens e.g. to restart the MCU*/
#define LV_ASSERT_HANDLER_INCLUDE <stdint.h>
#define LV_ASSERT_HANDLER while(1);   /*Halt by default*/

/*-------------
 * Debug
 *-----------*/

/*1: Draw random colored rectangles over the redrawn areas*/
#define LV_USE_REFR_DEBUG 0

/*1: Draw a red overlay for ARGB layers and a green overlay for RGB layers*/
#define LV_USE_LAYER_DEBUG 0

/*1: Draw overlays with different colors for each draw_unit's tasks.
 *Also add the index number of the draw unit on white background.
 *For layers add the index number of the draw unit on black background.*/
#define LV_USE_PARALLEL_DRAW_DEBUG 0

/*-------------
 * Others
 *-----------*/

#define LV_ENABLE_GLOBAL_CUSTOM 0
#if LV_ENABLE_GLOBAL_CUSTOM
    /*Header to include for the custom 'lv_global' function"*/
    #define LV_GLOBAL_CUSTOM_INCLUDE <stdint.h>
#endif

/*Default cache size in bytes.
 *Used by image decoders such as `lv_lodepng` to keep the decoded image in the memory.
 *If size is not set to 0, the decoder will fail to decode when the cache is full.
 *If size is 0, the cache function is not enabled and the decoded mem will be released immediately after use.*/
#define LV_CACHE_DEF_SIZE       0

/*Default number of image header cache entries. The cache is used to store the headers of images
 *The main logic is like `LV_CACHE_DEF_SIZE` but for image headers.*/
#define LV_IMAGE_HEADER_CACHE_DEF_CNT 0

/*Number of stops allowed per gradient. Increase this to allow more stops.
 *This adds (sizeof(lv_color_t) + 1) bytes per additional stop*/
#define LV_GRADIENT_MAX_STOPS   2

/* Adjust color mix functions rounding. GPUs might calculate color mix (blending) differently.
 * 0: round down, 64: round up from x.75, 128: round up from half, 192: round up from x.25, 254: round up */
#define LV_COLOR_MIX_ROUND_OFS  0

/* Add 2 x 32 bit variables to each lv_obj_t to speed up getting style properties */
#define LV_OBJ_STYLE_CACHE      0

/* Add `id` field to `lv_obj_t` */
#define LV_USE_OBJ_ID           0

/* Use lvgl builtin method for obj ID */
#define LV_USE_OBJ_ID_BUILTIN   0

/*Use obj property set/get API*/
#define LV_USE_OBJ_PROPERTY 0

/* VG-Lite Simulator */
/*Requires: LV_USE_THORVG_INTERNAL or LV_USE_THORVG_EXTERNAL */
#define LV_USE_VG_LITE_THORVG  0

#if LV_USE_VG_LITE_THORVG

    /*Enable LVGL's blend mode support*/
    #define LV_VG_LITE_THORVG_LVGL_BLEND_SUPPORT 0

    /*Enable YUV color format support*/
    #define LV_VG_LITE_THORVG_YUV_SUPPORT 0

    /*Enable 16 pixels alignment*/
    #define LV_VG_LITE_THORVG_16PIXELS_ALIGN 1

    /*Enable multi-thread render*/
    #define LV_VG_LITE_THORVG_THREAD_RENDER 0

#endif

/*=====================
 *  COMPILER SETTINGS
 *====================*/

/*For big endian systems set to 1*/
#define LV_BIG_ENDIAN_SYSTEM 0

/*Define a custom attribute to `lv_tick_inc` function*/
#define LV_ATTRIBUTE_TICK_INC

/*Define a custom attribute to `lv_timer_handler` function*/
#define LV_ATTRIBUTE_TIMER_HANDLER

/*Define a custom attribute to `lv_display_flush_ready` function*/
#define LV_ATTRIBUTE_FLUSH_READY

/*Required alignment size for buffers*/
#define LV_ATTRIBUTE_MEM_ALIGN_SIZE 1

/*Will be added where memories needs to be aligned (with -Os data might not be aligned to boundary by default).
 * E.g. __attribute__((aligned(4)))*/
#define LV_ATTRIBUTE_MEM_ALIGN

/*Attribute to mark large constant arrays for example font's bitmaps*/
#define LV_ATTRIBUTE_LARGE_CONST PROGMEM

/*Compiler prefix for a big array declaration in RAM*/
#define LV_ATTRIBUTE_LARGE_RAM_ARRAY EXTMEM
/*Place performance critical functions into a faster memory (e.g RAM)*/
#define LV_ATTRIBUTE_FAST_MEM

/*Export integer constant to binding. This macro is used with constants in the form of LV_<CONST> that
 *should also appear on LVGL binding API such as Micropython.*/
#define LV_EXPORT_CONST_INT(int_value) struct _silence_gcc_warning /*The default value just prevents GCC warning*/

/*Prefix all global extern data with this*/
#define LV_ATTRIBUTE_EXTERN_DATA

/* Use `float` as `lv_value_precise_t` */
#define LV_USE_FLOAT            0

/*==================
 *   FONT USAGE
 *===================*/

/*Montserrat fonts with ASCII range and some symbols using bpp = 4
 *https://fonts.google.com/specimen/Montserrat*/
#define LV_FONT_MONTSERRAT_8  0
#define LV_FONT_MONTSERRAT_10 0
#define LV_FONT_MONTSERRAT_12 1
#define LV_FONT_MONTSERRAT_14 1
#define LV_FONT_MONTSERRAT_16 1
#define LV_FONT_MONTSERRAT_18 0
#define LV_FONT_MONTSERRAT_20 0
#define LV_FONT_MONTSERRAT_22 0
#define LV_FONT_MONTSERRAT_24 0
#define LV_FONT_MONTSERRAT_26 0
#define LV_FONT_MONTSERRAT_28 0
#define LV_FONT_MONTSERRAT_30 0
#define LV_FONT_MONTSERRAT_32 0
#define LV_FONT_MONTSERRAT_34 0
#define LV_FONT_MONTSERRAT_36 0
#define LV_FONT_MONTSERRAT_38 0
#define LV_FONT_MONTSERRAT_40 0
#define LV_FONT_MONTSERRAT_42 0
#define LV_FONT_MONTSERRAT_44 0
#define LV_FONT_MONTSERRAT_46 0
#define LV_FONT_MONTSERRAT_48 0

/*Demonstrate special features*/
#define LV_FONT_MONTSERRAT_28_COMPRESSED 0  /*bpp = 3*/
#define LV_FONT_DEJAVU_16_PERSIAN_HEBREW 0  /*Hebrew, Arabic, Persian letters and all their forms*/
#define LV_FONT_SIMSUN_16_CJK            0  /*1000 most common CJK radicals*/

/*Pixel perfect monospace fonts*/
#define LV_FONT_UNSCII_8  0
#define LV_FONT_UNSCII_16 0

/*Optionally declare custom fonts here.
 *You can use these fonts as default font too and they will be available globally.
 *E.g. #define LV_FONT_CUSTOM_DECLARE   LV_FONT_DECLARE(my_font_1) LV_FONT_DECLARE(my_font_2)*/
#define LV_FONT_CUSTOM_DECLARE

/*Always set a default font*/
#define LV_FONT_DEFAULT &lv_font_montserrat_14

/*Enable handling large font and/or fonts with a lot of characters.
 *The limit depends on the font size, font face and bpp.
 *Compiler error will be triggered if a font needs it.*/
#define LV_FONT_FMT_TXT_LARGE 0

/*Enables/disables support for compressed fonts.*/
#define LV_USE_FONT_COMPRESSED 0

/*Enable drawing placeholders when glyph dsc is not found*/
#define LV_USE_FONT_PLACEHOLDER 1

/*=================
 *  TEXT SETTINGS
 *=================*/

/**
 * Select a character encoding for strings.
 * Your IDE or editor should have the same character encoding
 * - LV_TXT_ENC_UTF8
 * - LV_TXT_ENC_ASCII
 */
#define LV_TXT_ENC LV_TXT_ENC_UTF8

/*Can break (wrap) texts on these chars*/
#define LV_TXT_BREAK_CHARS " ,.;:-_)]}"

/*If a word is at least this long, will break wherever "prettiest"
 *To disable, set to a value <= 0*/
#define LV_TXT_LINE_BREAK_LONG_LEN 0

/*Minimum number of characters in a long word to put on a line before a break.
 *Depends on LV_TXT_LINE_BREAK_LONG_LEN.*/
#define LV_TXT_LINE_BREAK_LONG_PRE_MIN_LEN 3

/*Minimum number of characters in a long word to put on a line after a break.
 *Depends on LV_TXT_LINE_BREAK_LONG_LEN.*/
#define LV_TXT_LINE_BREAK_LONG_POST_MIN_LEN 3

/*Support bidirectional texts. Allows mixing Left-to-Right and Right-to-Left texts.
 *The direction will be processed according to the Unicode Bidirectional Algorithm:
 *https://www.w3.org/International/articles/inline-bidi-markup/uba-basics*/
#define LV_USE_BIDI 0
#if LV_USE_BIDI
    /*Set the default direction. Supported values:
    *`LV_BASE_DIR_LTR` Left-to-Right
    *`LV_BASE_DIR_RTL` Right-to-Left
    *`LV_BASE_DIR_AUTO` detect texts base direction*/
    #define LV_BIDI_BASE_DIR_DEF LV_BASE_DIR_AUTO
#endif

/*Enable Arabic/Persian processing
 *In these languages characters should be replaced with an other form based on their position in the text*/
#define LV_USE_ARABIC_PERSIAN_CHARS 0

/*==================
 * WIDGETS
 *================*/

/*Documentation of the widgets: https://docs.lvgl.io/latest/en/html/widgets/index.html*/

#define LV_WIDGETS_HAS_DEFAULT_VALUE  1

#define LV_USE_ANIMIMG    1

#define LV_USE_ARC        1

#define LV_USE_BAR        1

#define LV_USE_BUTTON        1

#define LV_USE_BUTTONMATRIX  1

#define LV_USE_CALENDAR   1
#if LV_USE_CALENDAR
    #define LV_CALENDAR_WEEK_STARTS_MONDAY 0
    #if LV_CALENDAR_WEEK_STARTS_MONDAY
        #define LV_CALENDAR_DEFAULT_DAY_NAMES {"Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"}
    #else
        #define LV_CALENDAR_DEFAULT_DAY_NAMES {"Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"}
    #endif

    #define LV_CALENDAR_DEFAULT_MONTH_NAMES {"January", "February", "March",  "April", "May",  "June", "July", "August", "September", "October", "November", "December"}
    #define LV_USE_CALENDAR_HEADER_ARROW 1
    #define LV_USE_CALENDAR_HEADER_DROPDOWN 1
#endif  /*LV_USE_CALENDAR*/

#define LV_USE_CANVAS     1

#define LV_USE_CHART      1

#define LV_USE_CHECKBOX   1

#define LV_USE_DROPDOWN   1   /*Requires: lv_label*/

#define LV_USE_IMAGE      1   /*Requires: lv_label*/

#define LV_USE_IMAGEBUTTON     1

#define LV_USE_KEYBOARD   1

#define LV_USE_LABEL      1
#if LV_USE_LABEL
    #define LV_LABEL_TEXT_SELECTION 1 /*Enable selecting text of the label*/
    #define LV_LABEL_LONG_TXT_HINT 1  /*Store some extra info in labels to speed up drawing of very long texts*/
    #define LV_LABEL_WAIT_CHAR_COUNT 3  /*The count of wait chart*/
#endif

#define LV_USE_LED        1

#define LV_USE_LINE       1

#define LV_USE_LIST       1

#define LV_USE_MENU       1

#define LV_USE_MSGBOX     1

#define LV_USE_ROLLER     1   /*Requires: lv_label*/

#define LV_USE_SCALE      1

#define LV_USE_SLIDER     1   /*Requires: lv_bar*/

#define LV_USE_SPAN       1
#if LV_USE_SPAN
    /*A line text can contain maximum num of span descriptor */
    #define LV_SPAN_SNIPPET_STACK_SIZE 64
#endif

#define LV_USE_SPINBOX    1

#define LV_USE_SPINNER    1

#define LV_USE_SWITCH     1

#define LV_USE_TEXTAREA   1   /*Requires: lv_label*/
#if LV_USE_TEXTAREA != 0
    #define LV_TEXTAREA_DEF_PWD_SHOW_TIME 1500    /*ms*/
#endif

#define LV_USE_TABLE      1

#define LV_USE_TABVIEW    1

#define LV_USE_TILEVIEW   1

#define LV_USE_WIN        1

/*==================
 * THEMES
 *==================*/

/*A simple, impressive and very complete theme*/
#define LV_USE_THEME_DEFAULT 1
#if LV_USE_THEME_DEFAULT

    /*0: Light mode; 1: Dark mode*/
    #define LV_THEME_DEFAULT_DARK 0

    /*1: Enable grow on press*/
    #define LV_THEME_DEFAULT_GROW 1

    /*Default transition time in [ms]*/
    #define LV_THEME_DEFAULT_TRANSITION_TIME 80
#endif /*LV_USE_THEME_DEFAULT*/

/*A very simple theme that is a good starting point for a custom theme*/
#define LV_USE_THEME_SIMPLE 1

/*A theme designed for monochrome displays*/
#define LV_USE_THEME_MONO 1

/*==================
 * LAYOUTS
 *==================*/

/*A layout similar to Flexbox in CSS.*/
#define LV_USE_FLEX 1

/*A layout similar to Grid in CSS.*/
#define LV_USE_GRID 1

/*====================
 * 3RD PARTS LIBRARIES
 *====================*/

/*File system interfaces for common APIs */

/*API for fopen, fread, etc*/
#define LV_USE_FS_STDIO 0
#if LV_USE_FS_STDIO
    #define LV_FS_STDIO_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_STDIO_PATH ""         /*Set the working directory. File/directory paths will be appended to it.*/
    #define LV_FS_STDIO_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

/*API for open, read, etc*/
#define LV_USE_FS_POSIX 0
#if LV_USE_FS_POSIX
    #define LV_FS_POSIX_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_POSIX_PATH ""         /*Set the working directory. File/directory paths will be appended to it.*/
    #define LV_FS_POSIX_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

/*API for CreateFile, ReadFile, etc*/
#define LV_USE_FS_WIN32 0
#if LV_USE_FS_WIN32
    #define LV_FS_WIN32_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_WIN32_PATH ""         /*Set the working directory. File/directory paths will be appended to it.*/
    #define LV_FS_WIN32_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

/*API for FATFS (needs to be added separately). Uses f_open, f_read, etc*/
#define LV_USE_FS_FATFS 0
#if LV_USE_FS_FATFS
    #define LV_FS_FATFS_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_FATFS_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

/*API for memory-mapped file access. */
#define LV_USE_FS_MEMFS 0
#if LV_USE_FS_MEMFS
    #define LV_FS_MEMFS_LETTER '\0'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
#endif

/*LODEPNG decoder library*/
#define LV_USE_LODEPNG 0

/*PNG decoder(libpng) library*/
#define LV_USE_LIBPNG 0

/*BMP decoder library*/
#define LV_USE_BMP 0

/* JPG + split JPG decoder library.
 * Split JPG is a custom format optimized for embedded systems. */
#define LV_USE_TJPGD 0

/* libjpeg-turbo decoder library.
 * Supports complete JPEG specifications and high-performance JPEG decoding. */
#define LV_USE_LIBJPEG_TURBO 0

/*GIF decoder library*/
#define LV_USE_GIF 0
#if LV_USE_GIF
/*GIF decoder accelerate*/
#define LV_GIF_CACHE_DECODE_DATA 0
#endif


/*Decode bin images to RAM*/
#define LV_BIN_DECODER_RAM_LOAD 0

/*RLE decompress library*/
#define LV_USE_RLE 0

/*QR code library*/
#define LV_USE_QRCODE 0

/*Barcode code library*/
#define LV_USE_BARCODE 0

/*FreeType library*/
#define LV_USE_FREETYPE 0
#if LV_USE_FREETYPE
    /*Memory used by FreeType to cache characters in kilobytes*/
    #define LV_FREETYPE_CACHE_SIZE 768

    /*Let FreeType to use LVGL memory and file porting*/
    #define LV_FREETYPE_USE_LVGL_PORT 0

    /* Maximum number of opened FT_Face/FT_Size objects managed by this cache instance. */
    /* (0:use system defaults) */
    #define LV_FREETYPE_CACHE_FT_FACES 8
    #define LV_FREETYPE_CACHE_FT_SIZES 8
    #define LV_FREETYPE_CACHE_FT_GLYPH_CNT 256
#endif

/* Built-in TTF decoder */
#define LV_USE_TINY_TTF 0
#if LV_USE_TINY_TTF
    /* Enable loading TTF data from files */
    #define LV_TINY_TTF_FILE_SUPPORT 0
#endif

/*Rlottie library*/
#define LV_USE_RLOTTIE 0

/*Enable Vector Graphic APIs*/
#define LV_USE_VECTOR_GRAPHIC  0

/* Enable ThorVG (vector graphics library) from the src/libs folder */
#define LV_USE_THORVG_INTERNAL 0

/* Enable ThorVG by assuming that its installed and linked to the project */
#define LV_USE_THORVG_EXTERNAL 0

/*Enable LZ4 compress/decompress lib*/
#define LV_USE_LZ4  0

/*Use lvgl built-in LZ4 lib*/
#define LV_USE_LZ4_INTERNAL  0

/*Use external LZ4 library*/
#define LV_USE_LZ4_EXTERNAL  0

/*FFmpeg library for image decoding and playing videos
 *Supports all major image formats so do not enable other image decoder with it*/
#define LV_USE_FFMPEG 0
#if LV_USE_FFMPEG
    /*Dump input information to stderr*/
    #define LV_FFMPEG_DUMP_FORMAT 0
#endif

/*==================
 * OTHERS
 *==================*/

/*1: Enable API to take snapshot for object*/
#define LV_USE_SNAPSHOT 0

/*1: Enable system monitor component*/
#define LV_USE_SYSMON   1
#if LV_USE_SYSMON
    /*Get the idle percentage. E.g. uint32_t my_get_idle(void);*/
    #define LV_SYSMON_GET_IDLE lv_timer_get_idle

    /*1: Show CPU usage and FPS count
     * Requires `LV_USE_SYSMON = 1`*/
    #define LV_USE_PERF_MONITOR 1
    #if LV_USE_PERF_MONITOR
        #define LV_USE_PERF_MONITOR_POS LV_ALIGN_BOTTOM_RIGHT

        /*0: Displays performance data on the screen, 1: Prints performance data using log.*/
        #define LV_USE_PERF_MONITOR_LOG_MODE 0
    #endif

    /*1: Show the used memory and the memory fragmentation
     * Requires `LV_USE_BUILTIN_MALLOC = 1`
     * Requires `LV_USE_SYSMON = 1`*/
    #define LV_USE_MEM_MONITOR 0
    #if LV_USE_MEM_MONITOR
        #define LV_USE_MEM_MONITOR_POS LV_ALIGN_BOTTOM_LEFT
    #endif

#endif /*LV_USE_SYSMON*/

/*1: Enable the runtime performance profiler*/
#define LV_USE_PROFILER 0
#if LV_USE_PROFILER
    /*1: Enable the built-in profiler*/
    #define LV_USE_PROFILER_BUILTIN 1
    #if LV_USE_PROFILER_BUILTIN
        /*Default profiler trace buffer size*/
        #define LV_PROFILER_BUILTIN_BUF_SIZE (16 * 1024)     /*[bytes]*/
    #endif

    /*Header to include for the profiler*/
    #define LV_PROFILER_INCLUDE "lvgl/src/misc/lv_profiler_builtin.h"

    /*Profiler start point function*/
    #define LV_PROFILER_BEGIN    LV_PROFILER_BUILTIN_BEGIN

    /*Profiler end point function*/
    #define LV_PROFILER_END      LV_PROFILER_BUILTIN_END

    /*Profiler start point function with custom tag*/
    #define LV_PROFILER_BEGIN_TAG LV_PROFILER_BUILTIN_BEGIN_TAG

    /*Profiler end point function with custom tag*/
    #define LV_PROFILER_END_TAG   LV_PROFILER_BUILTIN_END_TAG
#endif

/*1: Enable Monkey test*/
#define LV_USE_MONKEY 0

/*1: Enable grid navigation*/
#define LV_USE_GRIDNAV 0

/*1: Enable lv_obj fragment*/
#define LV_USE_FRAGMENT 0

/*1: Support using images as font in label or span widgets */
#define LV_USE_IMGFONT 0

/*1: Enable an observer pattern implementation*/
#define LV_USE_OBSERVER 1

/*1: Enable Pinyin input method*/
/*Requires: lv_keyboard*/
#define LV_USE_IME_PINYIN 0
#if LV_USE_IME_PINYIN
    /*1: Use default thesaurus*/
    /*If you do not use the default thesaurus, be sure to use `lv_ime_pinyin` after setting the thesauruss*/
    #define LV_IME_PINYIN_USE_DEFAULT_DICT 1
    /*Set the maximum number of candidate panels that can be displayed*/
    /*This needs to be adjusted according to the size of the screen*/
    #define LV_IME_PINYIN_CAND_TEXT_NUM 6

    /*Use 9 key input(k9)*/
    #define LV_IME_PINYIN_USE_K9_MODE      1
    #if LV_IME_PINYIN_USE_K9_MODE == 1
        #define LV_IME_PINYIN_K9_CAND_TEXT_NUM 3
    #endif /*LV_IME_PINYIN_USE_K9_MODE*/
#endif

/*1: Enable file explorer*/
/*Requires: lv_table*/
#define LV_USE_FILE_EXPLORER                     0
#if LV_USE_FILE_EXPLORER
    /*Maximum length of path*/
    #define LV_FILE_EXPLORER_PATH_MAX_LEN        (128)
    /*Quick access bar, 1:use, 0:not use*/
    /*Requires: lv_list*/
    #define LV_FILE_EXPLORER_QUICK_ACCESS        1
#endif

/*==================
 * DEVICES
 *==================*/

/*Use SDL to open window on PC and handle mouse and keyboard*/
#define LV_USE_SDL              0
#if LV_USE_SDL
    #define LV_SDL_INCLUDE_PATH    <SDL2/SDL.h>
    #define LV_SDL_RENDER_MODE     LV_DISPLAY_RENDER_MODE_DIRECT   /*LV_DISPLAY_RENDER_MODE_DIRECT is recommended for best performance*/
    #define LV_SDL_BUF_COUNT       1    /*1 or 2*/
    #define LV_SDL_FULLSCREEN      0    /*1: Make the window full screen by default*/
    #define LV_SDL_DIRECT_EXIT     1    /*1: Exit the application when all SDL windows are closed*/
#endif

/*Use X11 to open window on Linux desktop and handle mouse and keyboard*/
#define LV_USE_X11              0
#if LV_USE_X11
    #define LV_X11_DIRECT_EXIT         1  /*Exit the application when all X11 windows have been closed*/
    #define LV_X11_DOUBLE_BUFFER       1  /*Use double buffers for endering*/
    /*select only 1 of the following render modes (LV_X11_RENDER_MODE_PARTIAL preferred!)*/
    #define LV_X11_RENDER_MODE_PARTIAL 1  /*Partial render mode (preferred)*/
    #define LV_X11_RENDER_MODE_DIRECT  0  /*direct render mode*/
    #define LV_X11_RENDER_MODE_FULL    0  /*Full render mode*/
#endif

/*Driver for /dev/fb*/
#define LV_USE_LINUX_FBDEV      0
#if LV_USE_LINUX_FBDEV
    #define LV_LINUX_FBDEV_BSD           0
    #define LV_LINUX_FBDEV_RENDER_MODE   LV_DISPLAY_RENDER_MODE_PARTIAL
    #define LV_LINUX_FBDEV_BUFFER_COUNT  0
    #define LV_LINUX_FBDEV_BUFFER_SIZE   60
#endif

/*Use Nuttx to open window and handle touchscreen*/
#define LV_USE_NUTTX    0

#if LV_USE_NUTTX
    #define LV_USE_NUTTX_LIBUV    0

    /*Use Nuttx custom init API to open window and handle touchscreen*/
    #define LV_USE_NUTTX_CUSTOM_INIT    0

    /*Driver for /dev/lcd*/
    #define LV_USE_NUTTX_LCD      0
    #if LV_USE_NUTTX_LCD
        #define LV_NUTTX_LCD_BUFFER_COUNT    0
        #define LV_NUTTX_LCD_BUFFER_SIZE     60
    #endif

    /*Driver for /dev/input*/
    #define LV_USE_NUTTX_TOUCHSCREEN    0

#endif

/*Driver for /dev/dri/card*/
#define LV_USE_LINUX_DRM        0

/*Interface for TFT_eSPI*/
#define LV_USE_TFT_ESPI         0

/*Driver for evdev input devices*/
#define LV_USE_EVDEV    0

/*Drivers for LCD devices connected via SPI/parallel port*/
#define LV_USE_ST7735		0
#define LV_USE_ST7789		0
#define LV_USE_ST7796		0
#define LV_USE_ILI9341		0

#define LV_USE_GENERIC_MIPI (LV_USE_ST7735 | LV_USE_ST7789 | LV_USE_ST7796 | LV_USE_ILI9341)

/* LVGL Windows backend */
#define LV_USE_WINDOWS    0

/*==================
* EXAMPLES
*==================*/

/*Enable the examples to be built with the library*/
#define LV_BUILD_EXAMPLES 0

/*===================
 * DEMO USAGE
 ====================*/

/*Show some widget. It might be required to increase `LV_MEM_SIZE` */
#define LV_USE_DEMO_WIDGETS 0
#if LV_USE_DEMO_WIDGETS
    #define LV_DEMO_WIDGETS_SLIDESHOW 0
#endif

/*Demonstrate the usage of encoder and keyboard*/
#define LV_USE_DEMO_KEYPAD_AND_ENCODER 0

/*Benchmark your system*/
#define LV_USE_DEMO_BENCHMARK 0

/*Render test for each primitives. Requires at least 480x272 display*/
#define LV_USE_DEMO_RENDER 0

/*Stress test for LVGL*/
#define LV_USE_DEMO_STRESS 0

/*Music player demo*/
#define LV_USE_DEMO_MUSIC 1
#if LV_USE_DEMO_MUSIC
    #define LV_DEMO_MUSIC_SQUARE    0
    #define LV_DEMO_MUSIC_LANDSCAPE 1
    #define LV_DEMO_MUSIC_ROUND     0
    #define LV_DEMO_MUSIC_LARGE     0
    #define LV_DEMO_MUSIC_AUTO_PLAY 1
#endif

/*Flex layout demo*/
#define LV_USE_DEMO_FLEX_LAYOUT     0

/*Smart-phone like multi-language demo*/
#define LV_USE_DEMO_MULTILANG       0

/*Widget transformation demo*/
#define LV_USE_DEMO_TRANSFORM       0

/*Demonstrate scroll settings*/
#define LV_USE_DEMO_SCROLL          0

/*Vector graphic demo*/
#define LV_USE_DEMO_VECTOR_GRAPHIC  0
/*--END OF LV_CONF_H--*/

#endif /*LV_CONF_H*/

#endif /*End of "Content enable"*/
```

--- a/lv_conf_template.h
+++ b/lv_conf_template.h
@@ -17,6 +17,11 @@
 #ifndef LV_CONF_H
 #define LV_CONF_H
 
+/*If you need to include anything here, do it inside the `__ASSEMBLY__` guard */
+#if  0 && defined(__ASSEMBLY__)
+#include "my_include.h"
+#endif
+
 /*====================
    COLOR SETTINGS
  *====================*/

--- a/src/lv_conf_internal.h
+++ b/src/lv_conf_internal.h
@@ -73,6 +73,11 @@
  * Start parsing lv_conf_template.h
  -----------------------------------*/
 
+/*If you need to include anything here, do it inside the `__ASSEMBLY__` guard */
+#if  0 && defined(__ASSEMBLY__)
+#include "my_include.h"
+#endif
+
 /*====================
    COLOR SETTINGS
  *====================*/

